synapse/app/homeserver.py

```diff
@@ -399,6 +416,8 @@ def setup(config_options):
     hs.get_datastore().start_profiling()
     hs.get_replication_layer().start_get_pdu_cache()
 
+    reactor.callWhenRunning(hs.post_startup_check)
+
```

Comment:
Would it make more sense to check this before daemonizing? This fits in more with the correct situation where bad config causes synapse to die early, allowing people to have some confidence that the thing isn't just going to fall over once its been fully started.

I guess the nuisance here is that it is more difficult to run queries on the database before starting the reactor?